crates/rolldown/src/module_finalizers/mod.rs

```diff
@@ -447,13 +447,18 @@ impl<'me, 'ast> ScopeHoistingFinalizer<'me, 'ast> {
     // construct `{ prop_name: () => returned, ... }`
     let mut arg_obj_expr = ast::ObjectExpression::dummy(self.alloc);
 
-    arg_obj_expr.properties.extend(self.ctx.linking_info.canonical_exports(false).map(
+    // After we have https://github.com/rolldown/rolldown/blob/d6d65f9080e427cd9feef56eb7a110fbcf6c1414/crates/rolldown/src/stages/generate_stage/chunk_optimizer.rs#L347-L354
+    // module namespace is included does not always means all exports are used.
+    arg_obj_expr.properties.extend(self.ctx.linking_info.canonical_exports(false).filter_map(
       |(export, resolved_export)| {
+        if !self.ctx.used_symbol_refs.contains(&resolved_export.symbol_ref) {
+          return None;
+        }
```

Comment:
```suggestion
    arg_obj_expr.properties.extend(self.ctx.linking_info.canonical_exports(false).filter_map(
      |(export, resolved_export)| {
        // Even if the module namespace is included, some exports may not be used due to `optimize_facade_dynamic_entry_chunks`
        // https://github.com/rolldown/rolldown/blob/d6d65f9080e427cd9feef56eb7a110fbcf6c1414/crates/rolldown/src/stages/generate_stage/chunk_optimizer.rs#L347-L354
        if !self.ctx.used_symbol_refs.contains(&resolved_export.symbol_ref) {
          return None;
        }
```
I think we should focus on describing the behavior rather than the change